frontend/appflowy_flutter/lib/user/application/auth/supabase_auth_service.dart

```diff
@@ -0,0 +1,91 @@
+import 'package:appflowy/user/application/auth/auth_service.dart';
+import 'package:appflowy_backend/log.dart';
+import 'package:appflowy_backend/protobuf/flowy-error/errors.pb.dart';
+import 'package:appflowy_backend/protobuf/flowy-user/user_profile.pbserver.dart';
+import 'package:dartz/dartz.dart';
+import 'package:supabase_flutter/supabase_flutter.dart';
+
+class SupabaseAuthService implements AuthService {
+  const SupabaseAuthService();
+
+  GoTrueClient get _auth => Supabase.instance.client.auth;
+
+  @override
+  Future<Either<FlowyError, UserProfilePB>> signUp({
+    required String name,
+    required String email,
+    required String password,
+  }) async {
+    final response = await _auth.signUp(
+      email: email,
+      password: password,
+    );
+    final user = response.user;
+    if (user == null) {
+      // TODO: handle error
+      return left(FlowyError());
+    }
+    return Right(user.toUserProfile());
+  }
+
+  @override
+  Future<Either<FlowyError, UserProfilePB>> signIn({
+    required String email,
+    required String password,
+  }) async {
+    try {
+      final user = await _auth
+          .signInWithPassword(
+            email: email,
+            password: password,
+          )
+          .then((value) => value.user);
```

Comment:
```suggestion
      final response = await _auth.signInWithPassword(
        email: email,
        password: password,
      );
      
      final user = response.user;
```
Again, just personal preference, doesn't actually matter..